pytorch_lightning/plugins/environments/lsf_environment.py

```diff
@@ -51,112 +54,152 @@ def __init__(self):
             )
         self._main_address = self._get_main_address()
         self._main_port = self._get_main_port()
-        log.debug(f"MASTER_ADDR: {self._main_address}")
-        log.debug(f"MASTER_PORT: {self._main_port}")
-
-    @property
-    def creates_processes_externally(self) -> bool:
-        return True
-
-    @property
-    def main_address(self) -> str:
-        """The main address is read from a list of hosts contained in the environment variable `LSB_HOSTS`."""
-        return self._main_address
-
-    @property
-    def main_port(self) -> int:
-        """The main port gets calculated from the LSF job ID."""
-        return self._main_port
+        self._local_rank = self._get_local_rank()
+        self._global_rank = self._get_global_rank()
+        self._world_size = self._get_world_size()
+        self._node_rank = self._get_node_rank()
+
+        # set environment variables needed for initializing torch distributed process group
```

Comment:
Let's move this within a private method.

Reply:
Is this what you wanted moved to a private method?

Reply:
Yes, I think thought we could move the master_address / master port to assignment to a method to clean a bit the code.

Reply:
Neat !